src/common/handler-error.ts

```diff
@@ -1,100 +1,201 @@
+   *
+   * @experimental
+   */
+  constructor(type: HandlerErrorType, message?: string | undefined, options?: HandlerErrorOptions) {
+    const actualMessage = message || `Handler error: ${type}`;
+
+    super(actualMessage, { cause: options?.cause });
+    this.type = type;
+    this.retryableOverride = options?.retryableOverride;
+  }
 
   /**
-   * Whether this error should be considered retryable. If not specified, retry behavior is determined from the error
-   * type. For example, INTERNAL is retryable by default unless specified otherwise.
+   * Wraps an error in a {@link HandlerError}.
+   *
+   * This is a convenience method to wrap an existing error into a {@link HandlerError}.
+   *
+   * @param type - The type of the error.
+   * @param cause - The cause of the error.
+   * @returns A new {@link HandlerError} instance wrapping the error.
    */
-  retryable?: boolean;
+  public static wrap(type: HandlerErrorType, cause: unknown): HandlerError {
+    return new HandlerError(type, undefined, { cause });
+  }
+
+  /**
+   * Whether this error is retryable.
+   *
+   * This differs from the {@link retryableOverride} property in that `retryable` takes into
+   * account the default behavior resulting from the error type, if no override is provided.
+   *
+   * @see {@link retryableOverride}.
+   */
+  public get retryable(): boolean {
+    if (typeof this.retryableOverride === "boolean") return this.retryableOverride;
+
+    switch (this.type) {
+      case "BAD_REQUEST":
+      case "UNAUTHENTICATED":
+      case "UNAUTHORIZED":
+      case "NOT_FOUND":
+      case "NOT_IMPLEMENTED":
+        return false;
+
+      case "UNAVAILABLE":
+      case "UPSTREAM_TIMEOUT":
+      case "RESOURCE_EXHAUSTED":
+      case "INTERNAL":
+        return true;
+
+      default: {
+        // Force a compile time error if missing a case
+        const _noMissingCase: never = this.type;
+        return true;
+      }
+    }
+  }
 }
 
+injectSymbolBasedInstanceOf(HandlerError, "HandlerError");
+
 /**
- * Options for constructing a {@link HandlerError} from an underlying cause, type, and a retryable flag.
+ * Options for constructing a {@link HandlerError}.
+ *
+ * @experimental
+ * @inline
  */
-export interface HandlerErrorCauseOptions extends ErrorOptions {
-  /** One of the predefined error types. */
-  type: HandlerErrorType;
+export interface HandlerErrorOptions {
+  /**
+   * Underlying cause of the error.
+   */
+  cause?: unknown;
 
   /**
-   * Whether this error should be considered retryable. If not specified, retry behavior is determined from the error
-   * type. For example, INTERNAL is retryable by default unless specified otherwise.
+   * Whether this error should be considered retryable.
+   *
+   * If not set, the retry behavior is determined from the error type.
+   * For example, by default, `INTERNAL` is retryable, but `UNAVAILABLE` is non-retryable.
    */
-  retryable?: boolean;
+  retryableOverride?: boolean | undefined;
```

Comment:
Should be equivalent, no?

```suggestion
  retryableOverride?: boolean;
```

Reply:
I got bitten by this a few times in the past.